### PR TITLE
Add setting of number_extended_events

### PR DESCRIPTION
**Pull request by mwregan2**
Related to [JP-3096](https://jira.stsci.edu/browse/JP-3096)

This PR fixes the population of the new EXTNCRS keyword, which was not being set when in single processing mode.

**Checklist**
- [x] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [x] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

**Review by kmacdonald-stsci — APPROVED** [review]

**Review by kmacdonald-stsci — COMMENTED** [review]
The changes look fine, but there are code failures and CI failures that need to be corrected.

File: CHANGES.rst [hunk 1]
```diff
@@ -7,6 +7,17 @@ Bug Fixes
 jump
 ~~~~
 
+- Added setting of number_extended_events for non-multiprocessing
+  mode. This is the value that is put into the header keyword EXTNCRS.
```
Comment (hbushouse) [inline CHANGES.rst:11]: Needs to have a PR number listed at the end of the entry. That's why the CI check for a change log entry is failing.

**Comment by hbushouse** [comment]
Don't understand why the one pytest that was unskipped is causing failures in the CI tests (claims it can't find the input data file, which is clearly present in the data directory), but runs fine when I test it locally. So just to get the CI tests to work I marked it to be skipped again.

**Review by hbushouse — APPROVED** [review]
All CI tests are passing now. So I approve.

**Review by ddavis-stsci — APPROVED** [review]
LGTM, this should be skipped for romancal and romancal regression tests pass with this update. 
https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/273/

**Comment by hbushouse** [comment]
OK, it looks like we're all in agreement. Merging now.

**Pull request merged**